src/lib/dnssd/ServiceNaming.cpp

```diff
@@ -141,6 +141,14 @@ CHIP_ERROR MakeServiceSubtype(char * buffer, size_t bufferLen, DiscoveryFilter s
     case DiscoveryFilterType::kInstanceName:
         requiredSize = snprintf(buffer, bufferLen, "%s", subtype.instanceName);
         break;
+    case DiscoveryFilterType::kJointFabricMode:
+        // 3-bit number
+        if (subtype.code >= 1 << 3)
+        {
+            return CHIP_ERROR_INVALID_ARGUMENT;
+        }
+        requiredSize = snprintf(buffer, bufferLen, "_J%u", static_cast<uint16_t>(subtype.code));
```

Comment:
I don't see this _J bit in the spec.  Where is it?

Reply:
"J" stands for "Joint Fabric" and follows the same convention used in Commissioner Discovery (Section 4.3.3), where the "_" prefix is followed by the first letter of the acronym. Are you okay with this approach, or do you have any suggestions for improvement?

Reply:
The spec lists which subtypes are advertised.  It does not list this subtype.

If you think this should be advertised, fix the spec to say so.  There are no "conventions" here.  The subtypes that are defined are in '4.3.1.3. Commissioning Subtypes" in te spec.

Reply:
Agreed, It is missing from the spec.  I will review the need for a _J subtype for the Joint Fabric open commissioning window advertisement.  The spec does define the JF key on the other hand.

Reply:
Yes, I am not saying the JF TXT bits are not in the spec, just that this _J part isn't.

Reply:
@bzbarsky-apple Agreed.  @vijs I do not understand why we need _J?  I believe the JF key is sufficient to meet the requirements, implementation and test concerns for JF commissioning.  If there reason we need the _J subtype please explain.

Reply:
Without _J, it's possible that the controller could initiate the Joint Commissioning flow on a device that doesn't support Joint Commissioning, since only service subtypes are filtered according to the current CHIP tool implementation. This can be discussed separately.

To unblock this PR, _J has been removed in the latest commit to align with the latest specification.

Reply:
Moving this conversation to https://github.com/project-chip/connectedhomeip/issues/39062 for further discussion.